lib/SILOptimizer/Mandatory/AddressLowering.cpp

```diff
@@ -2776,6 +2776,13 @@ void UseRewriter::visitBeginBorrowInst(BeginBorrowInst *borrow) {
       allocStack->setIsLexical();
       return;
     }
+    auto storage = AccessStorageWithBase::compute(address);
```

Comment:
If you're not interested in AccessStorage, you should be able to use `getAccessBase` or `AccessBase::compute`

Reply:
Oh, perfect!  Changed to `getAccessBase` and checked propagated the lexical flag to it if its an `alloc_stack` and allowed it if it was a function argument.  Added a struct extract test case for both an opaque value that needed an alloc_stack and one that was passed in.